definitions/npm/react-redux_v4.x.x/flow_v0.30.x-/react-redux_v4.x.x.js

```diff
@@ -1,6 +1,6 @@
 /* @flow */
-type ConnectAll = <D, P, S, C: React$Component<D, P, S>, SP, DP, Dispatch: Function>(
-  mapStateToProps: (state: Object, ownProps: $Diff<$Diff<$Diff<P, DP>, SP>, D>) => SP,
+type ConnectAll = <D, P, S, C: React$Component<D, P, S>, SP, DP, Dispatch: Function, NP:$Diff<P, {dispatch: Dispatch} & SP & D>>(
+  mapStateToProps: (state: Object, ownProps: NP) => SP,
   mapDispatchToProps: (dispatch: Dispatch, ownProps: $Diff<$Diff<$Diff<P, DP>, SP>, D>) => DP,
```

Comment:
ownProps should probably be `NP` here as well if I'm following your other change.